testing/python/approx.py

```diff
@@ -582,3 +584,37 @@ def __len__(self):
 
         expected = MySizedIterable()
         assert [1, 2, 3, 4] == approx(expected)
+
+    def test_duck_typing(self):
+        """
+        Check that approx() works for objects which implemented the required
+        numeric methods (#8132).
+        """
+
+        @attr.s(auto_attribs=True)
+        class Container:
+            value: float
+
+            def __abs__(self) -> float:
+                return abs(self.value)
+
+            def __sub__(self, other):
```

Comment:
I did try type-annotating the return value like this:

```suggestion
            def __sub__(self, other: Any) -> Union[float, "Container"]:
```

However mypy complained with:

```
testing/python/approx.py:601:59: F821 undefined name 'Container'
testing/python/approx.py:608:60: F821 undefined name 'Container'
```

Is that because it is an inner class? Suggestions welcome. 👍